### PR TITLE
refactor(apollo-biz): Use constructor injection instead of field injection

### PR DESCRIPTION
**Pull request by klboke**
## What's the purpose of this PR

- refactor(apollo-biz): Use constructor injection instead of field injection

### motivation
- This PR is a continuation of #1839, the motivation for which is stated as follows in the quote below.

> This patch uses Constructor Injection to avoid [the drawbacks of Field Injection](https://www.google.com.hk/search?client=safari&rls=en&q=spring+field+injection&ie=UTF-8&oe=UTF-8&gws_rd=cr). What's more, Spring Documentation also recommends using Constructor Injection [here](https://docs.spring.io/spring/docs/4.2.x/spring-framework-reference/html/beans.html#beans-setter-injection)
> 
> Here are some of the drawbacks of Field Injection listed in [this article](https://blog.marcnuri.com/field-injection-is-not-recommended/):
> 
> Disallows immutable field declaration
> 
> Eases single responsibility principle violation
> 
> Tightly coupled with dependency injection container
> 
> Hidden dependencies

### CODE REVIEW

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/PortalDBPropertySource.java [hunk 1]
```diff
@@ -38,15 +38,18 @@
...
-  public PortalDBPropertySource(String name, Map<String, Object> source) {
+  public PortalDBPropertySource(final String name,
+      final Map<String, Object> source,
+      final ServerConfigRepository serverConfigRepository) {
     super(name, source);
+    this.serverConfigRepository = serverConfigRepository;
   }
-
-  public PortalDBPropertySource() {
+  @Autowired
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/PortalDBPropertySource.java:49]: Is this `@Autowired` required?

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/PortalDBPropertySource.java [hunk 2]
```diff
@@ -38,15 +38,18 @@
...
+  public PortalDBPropertySource( final ServerConfigRepository serverConfigRepository) {
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/PortalDBPropertySource.java:50]:
```suggestion
  public PortalDBPropertySource(final ServerConfigRepository serverConfigRepository) {
```

File: apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/BizDBPropertySource.java [hunk 3]
```diff
@@ -41,15 +41,19 @@ public class BizDBPropertySource extends RefreshablePropertySource {
...
-  public BizDBPropertySource(String name, Map<String, Object> source) {
+  public BizDBPropertySource(final String name,
+      final Map<String, Object> source,
+      final ServerConfigRepository serverConfigRepository) {
     super(name, source);
+    this.serverConfigRepository = serverConfigRepository;
   }
 
-  public BizDBPropertySource() {
+  @Autowired
```
Comment (nobodyiam) [inline apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/BizDBPropertySource.java:53]: Is the `@Autowired` required?

**Comment by nobodyiam** [comment]
Please also help to fix the conflict.

**Comment by klboke** [comment]
> Please also help to fix the conflict.

ok

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**